### PR TITLE
Fix overwhelming Blacklight deprecation warnings

### PR DESCRIPTION
**Pull request by jefferya**
Addresses #701, deprecation warnings appearing in production server logs 

Not meant to address deprecation warnings from test infrastructure

### CODE REVIEW

File: lib/avalon/m3u8_reader.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def initialize(playlist, base = '', recursive: true)
       @base = base
       @playlist = { files: [], playlists: [] }
       tags = {}
-      playlist.lines.each do |l|
+      playlist.each_line do |l|
```
Comment (jefferya) [inline lib/avalon/m3u8_reader.rb:40]: Addresses log message: StringIO#lines is deprecated; use #each_line instead

File: app/views/catalog/_index_media_object.html.erb [hunk 2]
```diff
@@ -21,7 +21,7 @@ Unless required by applicable law or agreed to in writing, software distributed
   <% index_fields(document).each do |field_name, field| -%>
     <% if should_render_index_field? document, field %>
 	    <dt class="blacklight-<%= field_name.parameterize %>"><%= render_index_field_label document, field: field_name %></dt>
-	    <dd class="blacklight-<%= field_name.parameterize %>"><%= doc_presenter.field_value field_name %></dd>
+	    <dd class="blacklight-<%= field_name.parameterize %>"><%= doc_presenter.field_value(doc_presenter.configuration.index_fields.fetch(field_name)) %></dd>
```
Comment (jefferya) [inline app/views/catalog/_index_media_object.html.erb:24]: Address: Blacklight 7 deprecation log message

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -28,6 +28,7 @@ class ApplicationController < ActionController::Base
 
   around_action :handle_api_request, if: proc{|c| request.format.json? || request.format.atom? }
   before_action :rewrite_v4_ids, if: proc{|c| request.method_symbol == :get && [params[:id], params[:content]].compact.any? { |i| i =~ /^[a-z]+:[0-9]+$/}}
+  skip_after_action :discard_flash_if_xhr # Suppress overwhelming Blacklight deprecation warning
```
Comment (jefferya) [inline app/controllers/application_controller.rb:31]: Address: Blacklight 7 deprecation log message

File: app/controllers/catalog_controller.rb [hunk 4]
```diff
@@ -180,6 +180,8 @@ class CatalogController < ApplicationController
     # If there are more than this many search results, no spelling ("did you
     # mean") suggestion is offered.
     config.spell_max = 5
+
+    config.fetch_many_document_params = { fl: "*" }
```
Comment (jefferya) [inline app/controllers/catalog_controller.rb:184]: Address: Blacklight 7 deprecation log message

File: app/views/catalog/_index_media_object.html.erb (hunk 2 above)
Comment (pgwillia) [reply app/views/catalog/_index_media_object.html.erb:24]: Is the deprecation about being a string when a symbol is required?  Could do `field_name.to_sym` instead?  But maybe I'm over simplifying the problem.

Comment (jefferya) [reply app/views/catalog/_index_media_object.html.erb:24]: I haven't gone far into the Blacklight codebase. The deprecation was on field_value's acceptance of a string argument and in Blacklight 7, field_value wants an instance of type StringBlacklight::Configuration::Field. I'll check the `field_name.to_sym`.

Comment (jefferya) [reply app/views/catalog/_index_media_object.html.erb:24]: Both `string` and `symbols` are deprecated with `field_value`.

**Review by pgwillia — APPROVED** [review]
LGTM 👍 Progress!

**Pull request merged**